src/main/java/org/datadog/jmxfetch/App.java

```diff
@@ -352,104 +366,189 @@ void start() {
     }
 
     public void doIteration() {
+        int numberOfMetrics = 0;
+        String instanceStatus = null;
+        String scStatus = null;
+        String instanceMessage = null;
```

Comment:
nit: I'd prefer if we initialized these variables closer to where they're used (i.e. in the first `for` loop)

Reply:
They're used in the catch, in case there's a problem with the executor pool, so they need to be outside of the `try{}` block.

Reply:
oh, I see, hmm I'm really nitpicking here but: in this case I wouldn't use the same variables in the top-level `catch {}` block. To me, for semantics, these variables should be initialized for each instance. Feel free to disregard though